.github/workflows/Tests.yaml

```diff
@@ -433,3 +429,88 @@ jobs:
           # We get occasional random timeouts, repeat tests to see if
           # it is a random timeout or systematic
           ctest -j2 --output-on-failure -L inputfiles --repeat after-timeout:3
+
+  arch_datastructures_tests:
+    name: Archs
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler:
+          - gcc-7
+          - gcc-9
+          - clang-6.0
+          - clang-10
+        include:
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake")
+            compiler: gcc-7
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server")
+            compiler: gcc-9
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake")
+            compiler: clang-6.0
+          - sde_arch: ("-nhm;nehalem" "-snb;sandybridge" "-hsw;haswell"
+              "-skl;skylake" "-icx;icelake-server" "-tgl;tigerlake")
+            compiler: clang-10
+    container:
+      image: sxscollaboration/spectrebuildenv:latest
+      env:
+        # See the unit test job for the reasons for these configuration choices
+        CXXFLAGS: "-Werror"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Configure, build, and run tests
+        working-directory: /work
+        # Notes on the build configuration:
+        # - We don't need debug symbols during CI, so we turn them off to reduce
+        #   memory usage.
+        run: >
+          ARCH_PARAM_LIST=${{ matrix.sde_arch }}
+
+          for ARCH_PARAM in ${ARCH_PARAM_LIST[@]}; do
+            OVERRIDE_ARCH=`echo ${ARCH_PARAM} | cut -d";" -f2`
+            SDE_FLAG=`echo ${ARCH_PARAM} | cut -d";" -f1`
+            echo "CMake arch flag: $OVERRIDE_ARCH"
+            echo "Intel SDE arch flag: $SDE_FLAG"
+            cd /work
+            BUILD_DIR=build$OVERRIDE_ARCH
+            mkdir $BUILD_DIR && cd $BUILD_DIR
+            if [[ ${{ matrix.compiler }} =~ (gcc|clang)-([0-9\.]+) ]]; then
+              CC=${BASH_REMATCH[1]}-${BASH_REMATCH[2]};
+              CHARM_CC=${BASH_REMATCH[1]};
+              if [[ ${BASH_REMATCH[1]} = gcc ]]; then
+                CXX=g++-${BASH_REMATCH[2]};
+                FC=gfortran-${BASH_REMATCH[2]};
+              else
+                CXX=clang++-${BASH_REMATCH[2]};
+                FC=gfortran-8;
+              fi
+            fi
+
+            cmake\
```

Comment:
you don't need the `\` line-continuations since the `>` in line 470 means that non-empty lines are concatenated

Reply:
This is incorrect for this particular scenario. I think it must be the for loop that prevents the usual automatic line continuation.

Reply:
hm, then I don't exactly understand how the bash and yaml syntax interact here, but if this works let's leave it as-is 🤷‍♂️